tokenizers/src/pre_tokenizers/unicode_scripts/pre_tokenizer.rs

```diff
@@ -0,0 +1,100 @@
+use serde::{Deserialize, Serialize};
+
+use crate::pre_tokenizers::unicode_scripts::scripts::{get_script, Script};
+use crate::tokenizer::{normalizer::Range, PreTokenizedString, PreTokenizer, Result};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(tag = "type")]
+pub struct UnicodeScripts {}
+
+impl UnicodeScripts {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for UnicodeScripts {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+// This code exists in the Unigram default IsValidSentencePiece.
+// It could be integrated directly within `get_script` but I
+// think it's kind of tricky to see those modifications later
+// I am guessing release mode will optimize this away anyway.
+fn fixed_script(c: char) -> Script {
+    let raw_script = get_script(c);
+    if c as u32 == 0x30FC {
+        Script::Han
+    } else {
+        match raw_script {
+            Script::Hiragana => Script::Han,
+            Script::Katakana => Script::Han,
+            script => script,
+        }
+    }
+}
+
+impl PreTokenizer for UnicodeScripts {
+    fn pre_tokenize(&self, pretokenized: &mut PreTokenizedString) -> Result<()> {
+        pretokenized.split(|_, normalized| {
+            let mut last_script = Script::Any;
```

Comment:
By doing so, are you sure that the offset 0 will always be the first in `ranges`?

Reply:
I am but I am going to go in favor of an Option for last_script. I guess that will be clearer to the reader.